gnovm/pkg/gnolang/machine.go

```diff
@@ -243,6 +273,47 @@ func (m *Machine) RunMemPackage(memPkg *std.MemPackage, save bool) (*PackageNode
 	return pn, pv
 }
 
+func filterDuplicates(fset *FileSet) {
+	defined := make(map[Name]struct{}, 128)
+	for _, f := range fset.Files {
+		for i := 0; i < len(f.Decls); i++ {
+			d := f.Decls[i]
+			var name Name
+			switch d := d.(type) {
+			case *FuncDecl:
+				name = d.Name
+				if d.IsMethod {
+					name = Name(derefStar(d.Recv.Type).String()) + "." + name
+				}
+			case *TypeDecl:
+				name = d.Name
+			case *ValueDecl:
+				if len(d.NameExprs) == 1 {
+					name = d.NameExprs[0].Name
+				} else {
+					// TODO: support multiple names
+					continue
+				}
```

Comment:
maybe is better to return an error?

Reply:
Implementing this condition in an effort to try and leave fewer scattered TODOs around the code :)